compiler/rustc_target/src/asm/mod.rs

```diff
@@ -712,3 +712,183 @@ pub fn allocatable_registers(
         }
     }
 }
+
+#[derive(
+    Copy,
+    Clone,
+    Encodable,
+    Decodable,
+    Debug,
+    Eq,
+    PartialEq,
+    PartialOrd,
+    Hash,
+    HashStable_Generic
+)]
+pub enum InlineAsmClobberAbi {
+    X86,
+    X86_64Win,
+    X86_64SysV,
+    Arm,
+    AArch64,
+    RiscV,
+}
+
+impl InlineAsmClobberAbi {
+    /// Parses a clobber ABI for the given target, or returns a list of supported
+    /// clobber ABIs for the target.
+    pub fn parse(
+        arch: InlineAsmArch,
+        target: &Target,
+        name: Symbol,
+    ) -> Result<Self, &'static [&'static str]> {
+        let name = &*name.as_str();
+        match arch {
+            InlineAsmArch::X86 => match name {
+                "C" | "cdecl" | "stdcall" | "fastcall" => Ok(InlineAsmClobberAbi::X86),
+                _ => Err(&["C", "cdecl", "stdcall", "fastcall"]),
+            },
+            InlineAsmArch::X86_64 => match name {
+                "C" if !target.is_like_windows => Ok(InlineAsmClobberAbi::X86_64SysV),
+                "C" if target.is_like_windows => Ok(InlineAsmClobberAbi::X86_64Win),
+                "win64" => Ok(InlineAsmClobberAbi::X86_64Win),
+                "sysv64" => Ok(InlineAsmClobberAbi::X86_64SysV),
+                _ => Err(&["C", "win64", "sysv64"]),
+            },
+            InlineAsmArch::Arm => match name {
+                "C" | "aapcs" => Ok(InlineAsmClobberAbi::Arm),
+                _ => Err(&["C", "aapcs"]),
+            },
+            InlineAsmArch::AArch64 => match name {
+                "C" => Ok(InlineAsmClobberAbi::AArch64),
```

Comment:
Why do we handle aapcs for ARM but not aarch64?

Reply:
On AArch64

```rust
pub extern "aapcs" fn aapcs_func() {}
```

results in:

```
LLVM ERROR: Unsupported calling convention.
```

Reply:
`aapcs` is only a valid calling convention on ARM, not AArch64.